File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        //check if minus 1 is in it 
        Set<Integer> hashset = new HashSet<>();
        for(int num: nums){
            hashset.add(num);
        }

        int max = 0; 
        for(int num : hashset){
            //go to beginning 
            if(!hashset.contains(num -1)){
                //1 num is 0
                int curr = num; 
                int sum = 0; 
                while(hashset.contains(curr )){
                    sum++;
                    curr ++;
                }

                max = Math.max(max, sum);
            }
        }

        return max;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-2.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int max = int.MinValue;
        int sum = 0;
        foreach(var num in nums){
            sum += num;
            max = Math.Max(max, sum);
            if(sum < 0) {
                sum = 0;
            }
            
        }

        return max;
    }
}
